### ISSUE TITLE
no-name-in-module while importing from mypy package

### ISSUE THREAD

**Issue by pbasista** [PyCQA/pylint#3451]
### Steps to reproduce

Run `pylint` on the following file:

```python
"Test of pylint on mypy imports."""
from mypy.types import Any

print(Any)
```

### Current behavior

```
test.py:2:0: E0611: No name 'Any' in module 'mypy.types' (no-name-in-module)
```

### Expected behavior

No error is reported. The module contains the `Any` identifier:

```python
>>> from mypy.types import Any
>>> print(Any)
typing.Any
```

But `pylint` for some reason does not recognize it. Its behavior on other modules from the `mypy` package is similar.

When I try to access the `__file__` attribute of the imported `mypy` module, it points to a binary file:

```
>>> import mypy.types
>>> mypy.types.__file__
'/home/peter/.virtualenvs/test/lib/python3.8/site-packages/mypy/types.cpython-38-x86_64-linux-gnu.so'
```

despite the fact that there is a corresponding Python module present in the same directory. I am not sure if that has anything to do with the `pylint`'s ability to recognize the available identifiers or not.

### pylint --version output

```
pylint 2.4.4
astroid 2.3.3
Python 3.8.2 (default, Feb 26 2020, 22:21:03) 
[GCC 9.2.1 20200130]
```

**Comment by PCManticore** [comment]
pylint cannot properly figure out the content of C extension, which I presume `mypy` is at this point. As such, there's not much `pylint` can do. You can try with `--extension-pkg-whitelist=mypy.types` but that seems to be crashing now for some reason. Alternatively, you can use `--generated-members=...` or `--ignored-modules=...` to completely ignore `mypy.types`.
A final solution would be to write an `astroid` brain tip https://github.com/PyCQA/astroid/tree/master/astroid/brain.

Hope that helps.

**Comment by pbasista** [comment]
Thank you for the information. I was unaware of these details.

> You can try with `--extension-pkg-whitelist=mypy.types` but that seems to be crashing now for some reason.

What do you mean by "crashing"? `pylint` itself does not crash for me, although the error message stays the same.

However, when I have tried it instead with `--extension-pkg-whitelist=mypy` (`mypy.types` is not a package but `mypy` is), then the error was no longer present. Also, other checks related to `mypy.types` module seem to work as expected. For example, when `pylint --extension-pkg-whitelist=mypy` is run on this file:

```python
"Test of pylint on mypy imports."""
from mypy.types import nonexistent_identifier

print(nonexistent_identifier)
```
it produces the following output:
```
% pylint --extension-pkg-whitelist=mypy test.py                                                                   :(
************* Module test
test.py:2:0: E0611: No name 'nonexistent_identifier' in module 'mypy.types' (no-name-in-module)

----------------------------------------------------------------------
Your code has been rated at -15.00/10 (previous run: -15.00/10, +0.00)
```

So, running `pylint` with `--extension-pkg-whitelist=mypy` parameter helps. Thank you for suggesting it.

Maybe it would be beneficial to have a guideline somewhere in FAQ saying that when a binary module cannot be checked, it needs to be added to this "extension whitelist". At least to me, it was not obvious how to resolve such an issue.

**Comment by PCManticore** [comment]
Sorry for the confusion, locally I had a recursion error while using `--extension-pkg-whitelist=mypy` and assumed it reproduces all the time.

The extension whitelist is documented here http://pylint.pycqa.org/en/latest/technical_reference/c_extensions.html but maybe it needs to be added in the FAQ as well.

**Comment by pbasista** [comment]
> locally I had a recursion error while using `--extension-pkg-whitelist=mypy`

I also get that error when using Python 3.6. With Python 3.7 or 3.8 it works correctly.
